nixos/modules/services/home-automation/homebridge.nix

```diff
@@ -0,0 +1,756 @@
+        # https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Arch-Linux
+        nodejs_22
+        nettools
+        gcc
+        gnumake
+        # Required for access to systemctl and journalctl
+        systemd
+        # Required for access to sudo
+        "/run/wrappers"
+        # Some plugins need bash to download tools
+        bash
+      ];
+
+      environment = {
+        HOMEBRIDGE_CONFIG_UI_TERMINAL = "1";
+        DISABLE_OPENCOLLECTIVE = "true";
+        # Required or homebridge will search the global npm namespace
+        UIX_STRICT_PLUGIN_RESOLUTION = "1";
+      };
+
+      # Settings found from standalone mode docs and hb-service code
+      # https://github.com/homebridge/homebridge-config-ui-x/wiki/Standalone-Mode
+      # https://github.com/homebridge/homebridge-config-ui-x/blob/a12ad881fe6df62d817ecb56f9fc6b7e82b6d078/src/bin/platforms/linux.ts#L714
+      serviceConfig = {
+        Type = "simple";
+        User = "homebridge";
+        PermissionsStartOnly = true;
+        WorkingDirectory = cfg.userStoragePath;
+        EnvironmentFile = mkIf (cfg.environmentFile != null) [ cfg.environmentFile ];
+        ExecStart = "${pkgs.homebridge-config-ui-x}/bin/homebridge-config-ui-x ${args}";
+        Restart = "always";
+        RestartSec = 3;
+        KillMode = "process";
+        CapabilityBoundingSet = [
+          "CAP_IPC_LOCK"
+          "CAP_NET_ADMIN"
+          "CAP_NET_BIND_SERVICE"
+          "CAP_NET_RAW"
+          "CAP_SETGID"
+          "CAP_SETUID"
+          "CAP_SYS_CHROOT"
+          "CAP_CHOWN"
+          "CAP_FOWNER"
+          "CAP_DAC_OVERRIDE"
+          "CAP_AUDIT_WRITE"
+          "CAP_SYS_ADMIN"
+        ];
+        AmbientCapabilities = [
+          "CAP_NET_RAW"
+          "CAP_NET_BIND_SERVICE"
+        ];
+      };
+    };
+
+    # Create a user whose home folder is the user storage path
+    users.users.homebridge = {
+      home = cfg.userStoragePath;
+      createHome = true;
+      group = "homebridge";
+      # Necessary so that this user can run journalctl
+      extraGroups = [ "systemd-journal" ];
+      isSystemUser = true;
+    };
+
+    users.groups.homebridge = { };
+
+    # Need passwordless sudo for a few commands
+    # homebridge-config-ui-x needs for some features
+    security.sudo.extraRules = [
+      {
+        users = [ "homebridge" ];
+        commands = [
+          {
+            # Ability to restart homebridge services
+            command = "${pkgs.systemd}/bin/systemctl restart homebridge homebridge-config-ui-x";
+            options = [ "NOPASSWD" ];
+          }
+          {
+            # Ability to shutdown server
+            command = "${pkgs.systemd}/bin/shutdown -h now";
+            options = [ "NOPASSWD" ];
+          }
+          {
+            # Ability to restart server
+            command = "${pkgs.systemd}/bin/shutdown -r now";
+            options = [ "NOPASSWD" ];
+          }
+        ];
+      }
+    ];
```

Comment:
This seems like it could potentially be dangerous, but unfortunately I'm not too certain of how this could be sandboxed. Although the impact here is pretty limited so it's probably (?) fine? You probably don't need to change anything.